homeassistant/components/suez_water/__init__.py

```diff
@@ -26,3 +32,47 @@ async def async_setup_entry(hass: HomeAssistant, entry: SuezWaterConfigEntry) ->
 async def async_unload_entry(hass: HomeAssistant, entry: SuezWaterConfigEntry) -> bool:
     """Unload a config entry."""
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def async_migrate_entry(
+    hass: HomeAssistant, config_entry: SuezWaterConfigEntry
+) -> bool:
+    """Migrate old suez water config entry."""
+    _LOGGER.debug(
+        "Migrating configuration from version %s.%s",
+        config_entry.version,
+        config_entry.minor_version,
+    )
+
+    if config_entry.version > SuezWaterConfigFlow.VERSION:
+        _LOGGER.error("Suez_water version downgrade not handled")
+        return False
+    if config_entry.version == 2:
+        _LOGGER.debug("Suez_water no minor changes in version 2.X")
+        return True
+
+    unique_id = config_entry.unique_id
+    if config_entry.version == 1:
+        # Going to 2.X
+        counter_id = config_entry.data.get(CONF_COUNTER_ID)
+        if not counter_id:
+            _LOGGER.error(
+                "Failed to migrate to suez_water because no counter_id was previously defined"
+            )
+            return False
+        unique_id = str(counter_id)
```

Comment:
I think counter_id is already a string?

Reply:
You are half right, your comment made me find a inconsistency the config flow schema request an optional string which if not found request the api which give a number.
I'll have to check which one should be used after

Reply:
I have added a migration for consistency